mock/py/mockbuild/util.py

```diff
@@ -1431,8 +1432,7 @@ def set_config_opts_per_cmdline(config_opts, options, args):
 
     check_config(config_opts)
     # can't cleanup unless resultdir is separate from the root dir
-    rootdir = os.path.join(config_opts['basedir'], config_opts['root'])
-    if is_in_dir(config_opts['resultdir'] % config_opts, rootdir):
+    if is_in_dir(config_opts['resultdir'] % config_opts, config_opts['rootdir']):
```

Comment:
This is wrong.  The previous `rootdir` variable isn`t `config_opts['rootdir']` but one level up.
As a result, the RPMs from result dir are removed by default (weird thing that the testsuite
did not catch this... so testsuite needs fix here as well).